src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ProcessTask.java

```diff
@@ -119,8 +117,38 @@ public ProcessTask(ShardInfo shardInfo, StreamConfig streamConfig, IRecordProces
                        RecordProcessorCheckpointer recordProcessorCheckpointer, KinesisDataFetcher dataFetcher,
                        long backoffTimeMillis, boolean skipShardSyncAtWorkerInitializationIfLeasesExist,
                        Optional<Integer> retryGetRecordsInSeconds, Optional<Integer> maxGetRecordsThreadPool) {
-        this(shardInfo, streamConfig, recordProcessor, recordProcessorCheckpointer, dataFetcher, backoffTimeMillis,
-                skipShardSyncAtWorkerInitializationIfLeasesExist,
+        this(shardInfo, streamConfig, recordProcessor, new SimpleRecordsFetcherFactory(streamConfig.getMaxRecords()),
```

Comment:
Why are you recreating the factory, try and create the factory only once. Trying passing the client lib config around instead of passing the factory.

Reply:
I have added new constructors to pass the config, I created new factory instance here because I thought some user might have direct dependency on this constructor, change the signature of this constructor might break their code.  If you think it is fine to get rid of this I can change it.